packages/realm/src/app-services/App.ts

```diff
@@ -206,46 +209,80 @@ export class App<FunctionsFactoryType = DefaultFunctionsFactory, CustomDataType
     return this.internal.config.appId;
   }
 
+  /**
+   * Log in a user.
+   */
   public async logIn(credentials: Credentials) {
     const userInternal = await this.internal.logInWithCredentials(credentials.internal);
     return User.get(userInternal, this);
   }
 
+  /**
+   * Perform operations related to the email/password auth provider.
+   */
   public get emailPasswordAuth(): EmailPasswordAuth {
     // TODO: Add memoization
     const internal = this.internal.usernamePasswordProviderClient();
     return new EmailPasswordAuth(internal);
   }
 
+  /**
+   * The last user to log in or being switched to.
+   */
   public get currentUser(): User<FunctionsFactoryType, CustomDataType> | null {
     const currentUser = this.internal.currentUser;
     return currentUser ? User.get(currentUser, this) : null;
   }
 
+  /**
+   * All authenticated users.
+   */
   public get allUsers(): Readonly<Record<string, User<FunctionsFactoryType, CustomDataType>>> {
     return Object.fromEntries(this.internal.allUsers.map((user) => [user.identity, User.get(user, this)]));
   }
 
+  /**
+   * Switch current user, from an instance of `User` or the string id of the user.
+   */
   public switchUser(): unknown {
     throw new Error("Not yet implemented");
   }
 
+  /**
+   * Logs out and removes a user from the client.
+   * @returns A promise that resolves once the user has been logged out and removed from the app.
+   */
   public async removeUser(user: AnyUser) {
     await this.internal.removeUser(user.internal);
   }
 
+  /**
+   * Delete the user.
+   * NOTE: This irrecoverably deletes the user from the device as well as the server!
+   * @returns A promise that resolves once the user has been deleted.
+   */
   public async deleteUser(user: AnyUser) {
     await this.internal.deleteUser(user.internal);
   }
 
+  /**
+   * Adds a listener that will be fired on various user events.
+   * This includes login, logout, switching users, linking users and refreshing custom data.
```

Comment:
```suggestion
   * This includes login, logout, switching users, linking users and refreshing custom data.
   * @param callback - Listener callback function
```

Reply:
It wasn't my intent to improve the comments as such. Just wanted to ensure we could delete the old types from the root without loosing anything valuable. I suggest leaving this to a future PR too.

Reply:
I understand that, but I was cleaning the campground while I was reading the review.